### PR TITLE
Add stack traces & support modules

### PR DESCRIPTION
**Pull request by bartversluijs**
### Problem
Runtime errors in imported jq modules showed incorrect file/line positions. Errors pointed to wrong lines (e.g., comment lines) because byte offsets from module AST nodes were interpreted against the main query's source text. There was no way to trace errors back through the call chain across modules.

#### Module source tracking
- Added `sourceInfo` and `pcPosition` types to associate each compiled instruction with its source file
- Changed `pcOffsets` from `map[int]int` to `map[int]pcPosition` to track which source file each byte offset belongs to
- Added `LoadModuleWithSource` interface so module loaders can provide the resolved file path and source text
- The built-in `moduleLoader` now resolves module paths to absolute paths

#### Error interfaces
- Added `SourcePositionError` interface — extends `PositionError` with `SourceFile()` and `SourceText()` for errors in imported modules
- Added `StackTraceError` interface — extends `SourcePositionError` with `CallStack() []StackFrame` for full call chain
- Added `StackFrame` public type with `Offset`, `SourceFile`, and `SourceText` fields

#### Stack trace collection
- Rewrote `wrapRuntimeError` to walk the scope stack and collect all caller frames with set-based deduplication
- Errors now carry a full call stack from the error location back to the outermost caller

#### Error formatting
- `FormatError` and `FormatErrorAt` now check `SourcePositionError` first, using the module's own source for line/column computation
- Stack traces are appended as `↳ file:line:col` lines
- All file paths are resolved to absolute paths via `filepath.Abs()`

#### Bogus `1:1` frame fix
- Compiler-generated `&Func{}` nodes (for `|=`, `_modify`, `recurse`, `debug`, format strings, etc.) now explicitly set `Offset: -1`, preventing phantom `1:1` stack frames from internal calls

### Files changed
- `compiler.go` — source tracking, `pcPosition` type, `LoadModuleWithSource` support, `Offset: -1` on internal `Func` nodes
- `error.go` — `runtimeError` extended with source file info and stack frames, implements new interfaces
- `execute.go` — `wrapRuntimeError` rewritten for stack trace collection with deduplication
- `env.go` — added `sources` and updated `pcOffsets` type
- `pos.go` — new public interfaces, `FormatError`/`FormatErrorAt` updated, `LineColumn`, `formatCallStack`, `absPath` helpers
- `module_loader.go` — added `LoadModuleWithSource` method
- `cli/cli.go` — absolute path for `-f` flag, stack frame rendering with `↳`
- `pos_test.go` — tests for all new functionality including offset-0 edge cases
- `cli/test.yaml` — updated/added CLI integration tests
- `cli/testdata/err_module.jq`, `cli/testdata/err_module2.jq` — test modules

### CODE REVIEW

File: pos.go [hunk 1]
```diff
@@ -57,11 +108,20 @@ type PositionError interface {
 //	    }
 //	}
 func FormatError(src string, err error) string {
-	if pe, ok := err.(PositionError); ok {
+	var msg string
+	if spe, ok := err.(SourcePositionError); ok && spe.SourceFile() != "" {
+		line, col := LineColumn(spe.SourceText(), spe.Position())
+		msg = fmt.Sprintf("%s:%d:%d: %s", absPath(spe.SourceFile()), line, col+1, err)
```
[the remark this answers is not in the document]
Comment (bartversluijs) [reply pos.go:118]: @copilot We want to have absolute paths. So that should be fine. Should any docs be updated?

File: execute.go [hunk 2]
```diff
@@ -374,24 +375,67 @@ func (env *env) wrapRuntimeError(err error, pc int) error {
...
-			return &runtimeError{err, offset}
+	if env.pcOffsets == nil {
+		return err
+	}
+
+	// Find the error position (where the error actually occurred).
+	var errorPos *pcPosition
+	if pos, ok := env.pcOffsets[pc]; ok {
+		errorPos = &pos
+	}
```
[the remark this answers is not in the document]
Comment (bartversluijs) [reply execute.go:388]: @copilot Please explain in detail. Does this fix the problem? Is it a workaround? What are we doing here?

**Comment by bartversluijs** [comment]
## Fix: Compile errors in imported JQ modules now report correct file and line

### Problem
When a compile error (like "variable not defined" or "function not defined") occurred inside
an imported JQ library module, the error only carried a byte offset via `PositionError`.
`FormatError`/`FormatErrorAt` then resolved that offset against the **main query's** source,
producing the wrong file name and line number.

For example, an undefined variable in `jqs/jq-lib.jq:122` would incorrectly be reported as
`jqs/lookup_menu_items.jq:23:1`.

### Root Cause
`variableNotFoundError` and `funcNotFoundError` implement `PositionError` (byte offset only),
but not `SourcePositionError` (which includes source file path and text). When these errors
bubble up from module compilation, no module context is attached.

### Fix
- **`error.go`** — Added `compileError` wrapper type implementing `SourcePositionError`,
  carrying the module's file name and source text alongside the original error.
- **`compiler.go`** — In `compileImport`, when a `PositionError` occurs during module
  compilation and module source info is available, wrap it in `compileError`.

### Test
- Added `TestSourcePositionError_CompileErrorInModule` covering both undefined variables
  and undefined functions in imported modules, verifying correct `SourceFile()`,
  `SourceText()`, line number, and `FormatErrorAt` output.

**Comment by bartversluijs** [comment]
Rebased commits into 2 commits, because of the Copilot stuff happening
